src/history/hs_rec.c

```diff
@@ -164,9 +164,10 @@ __hs_insert_record(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BTREE *btree,
                          * not idempotent, having them inserted multiple times can cause invalid
                          * values to be read.
                          */
-                        WT_ASSERT(session,
+                        WT_ASSERT_ALWAYS(session,
                           type != WT_UPDATE_MODIFY &&
-                            (uint8_t)upd_type_full_diag != WT_UPDATE_MODIFY);
+                            (uint8_t)upd_type_full_diag != WT_UPDATE_MODIFY,
+                          "Inserted invalid duplicate record of type modify update");
```

Comment:
Maybe: "Duplicate modifies inserted into the history store result in invalid reads."

Reply:
Its hard to gauge what level of detail is useful here, I think the original comment is pretty good as well.

Reply:
It feels like this assert is checking that the duplicate HS records aren't modify updates since this _can_ lead to reading invalid values. Do you think the suggested assertion message makes it sound as though the invalid read has already occurred instead of verifying the dup record type?

Reply:
Yeah that's a good point. Let's go with Luke's suggestion above

Reply:
In that case: "Duplicate modifies inserted into the history store can result in invalid reads."